connector/spanmetricsconnector/connector_test.go

```diff
@@ -676,22 +792,32 @@ func BenchmarkConnectorConsumeTraces(b *testing.B) {
 	}
 }
 
-func newConnectorImp(mcon consumer.Metrics, defaultNullValue *pcommon.Value, temporality string, logger *zap.Logger, ticker *clock.Ticker) *connectorImp {
+func newConnectorImp(
+	mcon consumer.Metrics,
+	defaultNullValue *pcommon.Value,
+	useExponentialHist bool,
+	temporality string,
+	logger *zap.Logger,
+	ticker *clock.Ticker,
+) *connectorImp {
 	defaultNotInSpanAttrVal := pcommon.NewValueStr("defaultNotInSpanAttrVal")
 	// use size 2 for LRU cache for testing purpose
-	metricKeyToDimensions, err := cache.NewCache[metricKey, pcommon.Map](DimensionsCacheSize)
+	metricKeyToDimensions, err := cache.NewCache[string, pcommon.Map](DimensionsCacheSize)
 	if err != nil {
 		panic(err)
 	}
+	histogramMetrics := initExplicitHistogramMetrics()
+	if useExponentialHist {
+		histogramMetrics = initExponentialHistogramMetrics()
+	}
```

Comment:
nit: what do you think about passing the constructor func as a param instead of `useExponentialHist`?